nixos/modules/programs/regreet.nix

```diff
@@ -0,0 +1,50 @@
+{ lib
+, pkgs
+, config
+, ...
+}:
+let
+  cfg = config.programs.regreet;
+  settingsType = pkgs.formats.toml { };
+in
+{
+  options.programs.regreet = {
+    enable = lib.mkEnableOption (lib.mdDoc "ReGreet, a greetd greeter");
+
+    package = lib.mkPackageOptionMD pkgs "regreet" { };
+
+    settings = lib.mkOption {
+      inherit (settingsType) type;
```

Comment:
Just a nitpick: I don't like the way this reads: `type (dot) type`. I would prefer `format (dot) type`. Indeed, I see many other modules using `settingsFormat` or `tomlFormat`, I personally would choose `tomlFormat` but I believe this is less common (and more unconventional) than the former.

Reply:
<https://github.com/NixOS/nixpkgs/blob/9e91539bb454af74b2e8f012e1aae08cb8abd9ff/nixos/modules/services/display-managers/greetd.nix#L7>